File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/CredentialStatus.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.bridge.proto.IdentityEntitiesV2;

/**
 * Represents the status information for a credential, including its ID, revocation nonce, and type.
 */
public class CredentialStatus {
  private final String id;
  private final long revocationNonce;
  private final String type;

  /**
   * Constructs an CredentialStatus object with the specified parameters.
   * @param id
   * @param revocationNonce
   * @param type
   */
  public CredentialStatus(String id, long revocationNonce, String type) {
    this.id = id;
    this.revocationNonce = revocationNonce;
    this.type = type;
  }

  public static CredentialStatus fromProto(IdentityEntitiesV2.CredentialStatusV2 res) {
    return new CredentialStatus(res.getId(), res.getRevocationNonce(), res.getType());
  }

  public IdentityEntitiesV2.CredentialStatusV2 toProto() {
    return IdentityEntitiesV2.CredentialStatusV2.newBuilder()
        .setId(this.id)
        .setRevocationNonce(this.revocationNonce)
        .setType(this.type)
        .build();
  }

  /**
   * Gets the ID associated with the credential.
   * @return
   */
  public String getId() {
    return id;
  }

  /**
   * Gets the revocation nonce associated with the credential.
   * @return
   */
  public long getRevocationNonce() {
    return revocationNonce;
  }

  /**
   * Gets the type of the credential.
   * @return
   */
  public String getType() {
    return type;
  }
}

File: languages/php/Bloock/Entity/IdentityV2/CredentialStatus.php
<?php

namespace Bloock\Entity\IdentityV2;

class CredentialStatus
{
    private string $id;
    private int $revocationNonce;
    private string $type;

    /**
     * @param string $id
     * @param int $revocationNonce
     * @param string $type
     */
    public function __construct(string $id, int $revocationNonce, string $type)
    {
        $this->id = $id;
        $this->revocationNonce = $revocationNonce;
        $this->type = $type;
    }

    public static function fromProto(\Bloock\CredentialStatusV2 $res): CredentialStatus
    {
        return new CredentialStatus($res->getId(), $res->getRevocationNonce(), $res->getType());
    }

    /**
     * @return string
     */
    public function getId(): string
    {
        return $this->id;
    }

    /**
     * @return int
     */
    public function getRevocationNonce(): int
    {
        return $this->revocationNonce;
    }

    /**
     * @return string
     */
    public function getType(): string
    {
        return $this->type;
    }

    public function toProto(): \Bloock\CredentialStatusV2
    {
        $p = new \Bloock\CredentialStatusV2();
        $p->setId($this->id);
        $p->setRevocationNonce($this->revocationNonce);
        $p->setType($this->type);
        return $p;
    }
}